### PR TITLE
Fixed Arrow Key Selection of Search Suggestion

### LINKED ISSUE #183: Add ability to select items from search autocomplete via keyboard
Now, when I want to select some autocomplete item, I need to click on it with a mouse. I want to have an option to stay on my keyboard, so the whole process is:

1. Search input is focused on the page load (e.g., `element.focus()`).
2. When I type something, autocomplete reveals.
3. I can use the Bottom and Top arrow keys to select a completed search query.
4. I press the Enter key, and it starts searching the selected query via the selected search engine (e.g., Google).

### PR DESCRIPTION
**Pull request by itz-rj-here**
Closes #183 
Bug fix requested by @ceskyDJ 

There is still a issue. I can't able to understand how can I use the same bg color of pointer hover color and the arrow key selection color. 
Here is the screenshot of the issue:-
image

Need some help on fixing this issue.

### CODE REVIEW

**Comment by itz-rj-here** [comment]
@XengShi Can you check if this works on other browsers until I try to fix the bg color please :o

**Comment by itz-rj-here** [comment]
Bruh chatgpt also giving wrong suggestion. 
He gave me this -_-

```
// Apply styles for active and hovered items
const style = document.createElement('style');
style.innerHTML = `
    .resultItem.active {
        background-color: #f0f0f0;  /* Change to the desired active background color */
        color: #0000ff;  /* Change to the desired active text color */
        cursor: pointer;  /* Change the cursor to pointer */
    }
    .resultItem:hover {
        background-color: #f0f0f0;  /* Change to the desired hover background color */
        color: #0000ff;  /* Change to the desired hover text color */
        cursor: pointer;  /* Change the cursor to pointer */
    }
`;
document.head.appendChild(style);

```

**Comment by itz-rj-here** [comment]
Any ideas guys?

**Comment by itz-rj-here** [comment]
@prem-k-r can you atleast help me on this selection color issue please 🥺

**Comment by prem-k-r** [comment]
ok done, i'll raise pr for it 

meanwhile, there is an issue, after visible result( first 5) it is not shifting/scrolling.. it is moving up/down but not visible

**Comment by itz-rj-here** [comment]
Ok let me look into it too.

**Comment by prem-k-r** [comment]
Also we ahve see if mouse cursor within resultBox, then keyboard selection unhighlighted

**Comment by XengShi** [comment]
> ok done, i'll raise pr for it 
> 
> meanwhile, there is an issue, after visible result( first 5) it is not shifting/scrolling.. it is moving up/down but not visible

I noticed one more bug.
When 'Ai tools' is enabled and opened/expended. If you disable the Ai tools toggle shortcuts just disappear. 

Hint :
When disabling Ai tools toggle it should first close or collapse [AI tools] button

**Comment by itz-rj-here** [comment]
Bruh It's not ready for review.

**Comment by itz-rj-here** [comment]
Someone of you need to fix the highlight thingy when we will let this pr to be merged.

**Comment by XengShi** [comment]
> @XengShi Can you check if this works on other browsers until I try to fix the bg color please :o

I will check tomorrow 💤😴

**Comment by XengShi** [comment]
> Bruh It's not ready for review.
> 

I accidentally touched that button 🙃 don't know how to ctrl+z

**Comment by prem-k-r** [comment]
> I noticed one more bug. When 'Ai tools' is enabled and opened/expended. If you disable the Ai tools toggle shortcuts just disappear.
> 
> Hint : When disabling Ai tools toggle it should first close or collapse [AI tools] button
> 
> 

The fold/unfold(little up/down arrow) botton might be affecting it.
I would say we should remove it, it's unnecessary.

**Comment by prem-k-r** [comment]
> if mouse cursor is within resultBox, then keyboard selection must be unhighlighted

okay, done this too.
@itz-rj-here please merge PR in your forked repo

**Comment by itz-rj-here** [comment]
When the suggestions are hovered, and using the arrow key the arrow key lets the cursor go to the first and last point of the text.

image

**Comment by itz-rj-here** [comment]
Check the chrome search bar. Use the pointer to hover and use the arrow key to highlight. We need to setup it like that.

**Comment by itz-rj-here** [comment]
Main thing is we need to make it like 2 text will be highlighten when selecting a text from search suggestion.

**Comment by prem-k-r** [comment]
Shouldn't having 2 highlighted text. Have to choose one. Prioritising mouse cursor over keyboard.

**Comment by prem-k-r** [comment]
We can do if mouse exits suggestion area not clicked outside, then remember the highlighted position and do up/down by keyboard from that position instead of first.

**Comment by prem-k-r** [comment]
Or Last detected movement from either of them

> Shouldn't having 2 highlighted text. Have to choose one. Prioritising mouse cursor over keyboard.

**Comment by itz-rj-here** [comment]
We need to fix that first then merge it.

File: style.css [hunk 1]
```diff
@@ -107,51 +107,53 @@
...
+ }
 
+.resultItem.active,
 .resultItem:hover {
-   background-color: var(--darkColor-blue);
-   border-radius: 15px;
-   color: #fff;
-}
+    background-color: var(--darkColor-blue);
+    border-radius: 15px;
```
Comment (prem-k-r) [inline style.css:153]: let's make it more round, 
`border-radius: 18px;`

**Review by prem-k-r — APPROVED** [review]
I think we are good to go already

File: style.css (hunk 1 above)
Comment (itz-rj-here) [reply style.css:153]: Ok

**Comment by itz-rj-here** [comment]
I personally don't liked the way the text is being highlighten. 

Check this: https://screenrec.com/share/84oACiDOdr

I'm trying to make it like this. :)

**Comment by itz-rj-here** [comment]
What you think should I try to do this or let this as it is?..

**Comment by prem-k-r** [comment]
I'll check it in few minutes

**Comment by itz-rj-here** [comment]
> I'll check it in few minutes

It's actually all on you if you think its ok then i will make this pr ready for review.

**Comment by prem-k-r** [comment]
https://github.com/user-attachments/assets/b92dcde6-cef6-42d4-b6e8-21737523d1e1

@itz-rj-here something like this?

**Comment by prem-k-r** [comment]
image

**Comment by prem-k-r** [comment]
> I think you should use a minimal symetric `border-radius` like Google does.

image
this?

**Comment by XengShi** [comment]
> > I think you should use a minimal symetric `border-radius` like Google does.
> 
> image
> this?

The current one is looking better

**Comment by prem-k-r** [comment]
ok, then let's go?

btw how about decreasing gap between searchbar and suggestion box
image

**Comment by itz-rj-here** [comment]
> ok, then let's go?
> 
> btw how about decreasing gap between searchbar and suggestion box
> image
> or this much:
> image
> 

The upper one is more better

**Comment by itz-rj-here** [comment]
> > > ok, then let's go?
> > > btw how about decreasing gap between searchbar and suggestion box
> > > image
> > > or this much:
> > > image
> > 
> > 
> > The upper one is more better
> 
> ok for it, delete `break` in html, and replace `top: 1;` with `bottom: -100px;` in css

Which break?
I think we should go with the old css style. Its way better than the new one.

**Comment by prem-k-r** [comment]
> > ok for it, delete `break` in html, and replace `top: 1;` with `bottom: -100px;` in css
> 
> Which break? I think we should go with the old css style. Its way better than the new one.

almost same though
old:
image
image

new:
image
image

**Comment by itz-rj-here** [comment]
I think we should go with it.
@ceskyDJ @XengShi

**Comment by itz-rj-here** [comment]
@ceskyDJ r u alive bruh 🗿?

**Comment by ceskyDJ** [comment]
> @ceskyDJ r u alive bruh 🗿?

I'm sorry, I'm just too busy with school. I think it's not that bad. I'd personally append the suggestion list just under the search input and don't use so big `border-radius` for highlighting selected search suggestions, but overall, it's pretty usable.

**Comment by itz-rj-here** [comment]
> > @ceskyDJ r u alive bruh 🗿?
> 
> I'm sorry, I'm just too busy with school. I think it's not that bad. I'd personally append the suggestion list just under the search input and don't use so big `border-radius` for highlighting selected search suggestions, but overall, it's pretty usable.

@prem-k-r pls check

**Comment by prem-k-r** [comment]
If we align it with search text, it doesn't looks good, background `Search with` is partially visible, we have to hide it then when suggestions are visible. 

For `border-radius: 16px;`, change it if needed, css line 153

**Comment by prem-k-r** [comment]
> > @ceskyDJ r u alive bruh 🗿?
> 
> I'm sorry, I'm just too busy with school. I think it's not that bad. I'd personally append the suggestion list just under the search input and don't use so big `border-radius` for highlighting selected search suggestions, but overall, it's pretty usable.

image

**Comment by prem-k-r** [comment]
image
shifting doesn't looks good to me, but we can change radius to this, if you all agree

**Comment by itz-rj-here** [comment]
> > image shifting doesn't looks good to me, but we can change radius to this, if you all agree
> 
> I like this one more,
> @itz-rj-here @XengShi, you?

Okayee do that.

**Comment by XengShi** [comment]
Yes, search suggestions width = searchbar's width is looking better

**Comment by prem-k-r** [comment]
yeah, but what about border radius, @ceskyDJ asked for smaller radius
this one?
image

 or what we already have?
image

**Comment by XengShi** [comment]
Just use one which matches perfectly with other elements of extension.

**Comment by prem-k-r** [comment]
1
image

 or  2
image

**Comment by itz-rj-here** [comment]
> 1
> image
> 
>  or  2
> image
> 

Actually 1 😅

**Comment by prem-k-r** [comment]
Ok

**Pull request merged**